public/app/plugins/panel/timeseries/plugins/ExemplarsPlugin.tsx

```diff
@@ -86,3 +127,69 @@ export const ExemplarsPlugin: React.FC<ExemplarsPluginProps> = ({ exemplars, tim
     />
   );
 };
+
+/**
+ * Function to get labels that are currently displayed in the legend
+ */
+export const getVisibleLabels = (
+  config: UPlotConfigBuilder,
+  frames: DataFrame[] | null
+): { labels: Labels[]; totalSeriesCount: number } => {
+  const visibleSeries = config.series.filter((series) => series.props.show);
+  const visibleLabels: Labels[] = [];
+  if (frames?.length) {
+    visibleSeries.forEach((plotInstance) => {
+      const frameIndex = plotInstance.props?.dataFrameFieldIndex?.frameIndex;
+      const fieldIndex = plotInstance.props?.dataFrameFieldIndex?.fieldIndex;
```

Comment:
Would they return undefined or `NaN` if optional chaining fails without reaching to `fieldIndex` or `frameIndex`

Reply:
should be [undefined](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Optional_chaining)